src/service/resources-impl.js

```diff
@@ -218,8 +229,11 @@ export class ResourcesImpl {
      */
     this.intersectionObserverCallbackFired_ = false;
 
+    this.divertIntersectResources_();
+
     if (
-      isExperimentOn(this.win, 'intersect-resources') &&
+      getExperimentBranch(this.win, INTERSECT_RESOURCES_EXP.id) ===
```

Comment:
One note: we don't know yet by this point whether we can run this experiment even if we select the branch. That will only happen if later we confirm that the API exists. So the numbers will be somewhat dirty. Is this ok? Or should we reverse and first test eligibility before throwing a coin?

Reply:
Yah, I think this needs to be done lower. Essentially, try to register the InOb. Then see if we're in the control branch, in which case we'd disconnect it and cleanup.

Reply:
The eligibility test is this:

```
function isEligible() {
  try {
    new IntersectionObserver(() => {}, {root: document});
    return true;
  } catch (e) {
    return false;
  }
}
```

Reply:
Thanks for the check logic. Switched to only pick branch if InOb supported.